src/main/java/io/odpf/firehose/filter/json/JsonFilter.java

```diff
@@ -0,0 +1,120 @@
+package io.odpf.firehose.filter.json;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import io.odpf.firehose.config.FilterConfig;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.filter.Filter;
+import io.odpf.firehose.filter.FilterException;
+import io.odpf.firehose.metrics.Instrumentation;
+import org.apache.commons.lang3.reflect.MethodUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static io.odpf.firehose.config.enums.FilterDataSourceType.KEY;
+import static io.odpf.firehose.config.enums.FilterDataSourceType.NONE;
+
+/**
+ * JSON-based filter to filter protobuf/JSON messages based on rules
+ * defined in a JSON Schema string.
+ */
+public class JsonFilter implements Filter {
```

Comment:
```

    private final Instrumentation instrumentation;
    private final JsonSchema schema;
    private final FilterConfig filterConfig;
    private final JsonFormat.Printer jsonPrinter = JsonFormat.printer().preservingProtoFieldNames();
    private final ObjectMapper objectMapper = new ObjectMapper();

    /**
     * Instantiates a new Json filter.
     *
     * @param filterConfig    the consumer config
     * @param instrumentation the instrumentation
     */
    public JsonFilter(FilterConfig filterConfig, Instrumentation instrumentation) {
        this.filterConfig = filterConfig;
        this.instrumentation = instrumentation;
        switch (filterConfig.getFilterJsonDataSource()) {
            case KEY:
            case MESSAGE:
                JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
                schema = schemaFactory.getSchema(filterConfig.getFilterJsonSchema());
                break;
            default:
                schema = null;
        }
    }
```

Reply:
if you have filterEnable config, it will be 
```
    public JsonFilter(FilterConfig filterConfig, Instrumentation instrumentation) {
        this.filterConfig = filterConfig;
        this.instrumentation = instrumentation;
        if (filterConfig.isFilterEnable()) {
            JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
            schema = schemaFactory.getSchema(filterConfig.getFilterJsonSchema());
        } else
            schema = null;
    }

```

Reply:
OR one you have fixed the implementation with NoOp class, then we don't need to check.. it will be covered in validation itself. 
```
    public JsonFilter(FilterConfig filterConfig, Instrumentation instrumentation) {
        this.filterConfig = filterConfig;
        this.instrumentation = instrumentation;
        JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
        schema = schemaFactory.getSchema(filterConfig.getFilterJsonSchema());
    }
```

Reply:
fixed